components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -86,7 +87,7 @@ private ClaimUtil() {
                     .getAuthorizationContextToken().getTokenString());
 
             // If the authenticated user is a federated user, no requirement to retrieve claims from local users tore.
-            if (accessTokenDO != null) {
+            if (!OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && accessTokenDO != null) {
```

Comment:
Better add a comment saying this is to consider the case of a federated user associated to a local user.